engine/src/main/java/org/terasology/rendering/primitives/BlockMeshGeneratorSingleShape.java

```diff
@@ -173,14 +177,16 @@ private boolean isSideVisibleForBlockTypes(Block blockToCheck, Block currentBloc
 
     @Override
     public Mesh getStandaloneMesh() {
-        if (mesh == null || mesh.isDisposed()) {
-            generateMesh();
+        Mesh m = this.mesh;
```

Comment:
This kind of optimization is really only meaningful when you have the instance field accessed in a loop.  I would kindly suggest that micro-optimizing these kind of changes is bad for readability because it adds another variable to keep track of in your head as you read the code.